capstone/src/main/java/com/google/univiz/scorecard/URLProviderImpl.java

```diff
@@ -14,6 +14,14 @@
 import javax.inject.Inject;
 
 final class URLProviderImpl implements URLProvider {
+  private static final String FRONT_URL =
+      "https://api.data.gov/ed/collegescorecard/v1/schools.json?";
+  private static final String QUERY_TYPE_NAME = "school.name=";
+  private static final String QUERY_TYPE_ID = "id=";
+  private static final String PER_PAGE = "&per_page=";
+  private static final String FIELDS_PARAM = "&fields=";
+  private static final String SUGGESTION_FIELDS = "id,school.name";
```

Comment:
By hard coding `SUGGESTION_FIELDS` we lose the generic nature of the reflection in the constructor. Can you think of a way to refactor this s.t. we can maintain the ability to inspect the fields from the representation classes in a reusable way?

Reply:
Hi Chris, thanks for your feedback. I'm a bit confused as to what you mean by refactoring-as of right now, these are the only fields that we want to get back from the colleges. What inspection are you thinking of?

Reply:
Right now, the fields to query when making the REST request are only based on the Gson annotations in `ScorecardData`. So if new fields need to be added, only `ScorecardData` would need to be updated.

If we hard code these fields for `SuggestionData` in this `SUGGESTION_FIELDS` constant, now the fields are replicated in two places.

Is there a way to prevent that? Can the logic in the constructor be used for both? Or even better - can it be used for a generic GSON representation?

Reply:
done :)